### ISSUE TITLE
XLIFF source strings not being interpreted correctly

### ISSUE THREAD

**Issue by ShadowXVII** [WeblateOrg/weblate#2481]
**Describe the bug**
If I use the following two files in a GIT repository, they don't seem to load correctly into Weblate.

**To Reproduce**

I have the following setup in a GIT repo;

- The monolingual file is `Base.g.xlf` that has all the source strings 
```
<?xml version="1.0" encoding="utf-8"?>
<xliff version="1.2" xmlns="urn:oasis:names:tc:xliff:document:1.2" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="urn:oasis:names:tc:xliff:document:1.2 xliff-core-1.2-transitional.xsd">
  <file original="Base" datatype="plaintext" source-language="en-US">
    <body>
      <group id="body">
        <trans-unit id="135956960462609535">
          <source>Color</source>
        </trans-unit>
      </group>
    </body>
  </file>
</xliff>
```

- The translations are in a sub-directory, /locale/*.xlf. I've a blank file, ready for the translation. For example, en-GB.xlf
```
<?xml version="1.0" encoding="utf-8"?>
<xliff version="1.2" xmlns="urn:oasis:names:tc:xliff:document:1.2" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="urn:oasis:names:tc:xliff:document:1.2 xliff-core-1.2-transitional.xsd">
  <file original="Base" datatype="plaintext" source-language="en-US" target-language="en-GB">
    <body>
    </body>
  </file>
</xliff>
```

**Expected behavior**
The source string should be populated, in this case with the word "Color".

**Screenshots**
![image](https://user-images.githubusercontent.com/446329/50872788-107c6500-1414-11e9-916d-2b7771cafcb1.png)

**Server configuration and status**
```
 * Weblate 3.3
 * Python 3.6.7
 * Django 2.1.3
 * Celery 4.2.1
 * celery-batches 0.2
 * six 1.11.0
 * social-auth-core 2.0.0
 * social-auth-app-django 3.1.0
 * django-appconf 1.0.2
 * translate-toolkit 2.3.1
 * Whoosh 2.7.4
 * defusedxml 0.5.0
 * Git 2.17.1
 * Pillow 5.1.0
 * python-dateutil 2.7.5
 * lxml 4.2.1
 * django-crispy-forms 1.7.2
 * django_compressor 2.2
 * djangorestframework 3.9.0
 * user-agents 1.1.0
 * jellyfish 0.6.1
 * pytz 2018.7
 * pyuca 1.2
 * PyYAML 3.12
 * tesserocr 2.3.1
 * Mercurial 4.5.3
 * git-svn 2.17.1
 * Database backends: django.db.backends.postgresql
 * Cache backends: avatar:FileBasedCache, default:RedisCache
 * Celery: redis://cache:6379/1, redis://cache:6379/1, regular
 * Platform: Linux 4.15.0-1035-azure (x86_64)
```

And a bunch of these in deploy check (not sure why, this is docker image)
```
?: (weblate.E002) Path /app/data/x is not writable, check your DATA_DIR settings.
        HINT: https://docs.weblate.org/en/weblate-3.3/admin/install.html#file-permissions
```

**Additional context**
I'm not sure if this is a XLIFF syntax error -- it's valid according to OpenXLIFF, and I've also spot checked it against the spec.

Using -'s in language codes, as per spec, discussed in #2480. Though likely unrelated.

**Comment by nijel** [comment]
It's probably bug in monolingual Xliff handling, the problem is that people have different expectations how monolingual Xliff should behave. Weblate currently matches the units by ID (or resname if available), but uses *target* from the monolingual file as source. This is working fine for cases where you have base language as English, where both source and target are set. For your case, it should probably use source instead.

**Comment by ShadowXVII** [comment]
Ah that would make sense then. Yes, it should fall back to Source, as Target isn't a required tag.

I wonder if that would have the knock-on effect of suggesting it's untranslated? As the Source and Target language are the same, I guess that would suggest it doesn't require a translation (and therefore monolingual).

Thanks for looking into it.

**Comment by nijel** [comment]
Thank you for your report, the issue you have reported has just been fixed. 

* In case you see problem with the fix, please comment on this issue. 
* In case you see similar problem, please open separate issue.
* If you are happy with the outcome, consider supporting Weblate by [donating](https://weblate.org/donate/).

**Comment by ShadowXVII** [comment]
Hi nijel,

This has worked in 3.4 for Translation languages. The source shows when editing en-AU, for example.

However when editing the base/source language (en-US), it doesn't appear -- wondering if there's a secondary issue here specific to the monolingual file?

In this file, the source and target are the same (en-US). This is the monolingual base file.

```
<?xml version="1.0" encoding="utf-8"?>
<xliff version="1.2" xmlns="urn:oasis:names:tc:xliff:document:1.2" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="urn:oasis:names:tc:xliff:document:1.2 xliff-core-1.2-transitional.xsd">
  <file datatype="xml" source-language="en-US" target-language="en-US" original="Translation">
    <body>
      <trans-unit id="123456789" size-unit="char" translate="yes" xml:space="preserve">
        <source>Translation</source>
      </trans-unit>
...
```

Thanks,
Jake

**Comment by nijel** [comment]
There is just the editable source shown when editing base file, see for example:

https://hosted.weblate.org/translate/androbd/strings/en/?checksum=2c6a6e1dcd3d1dbb

**Comment by ShadowXVII** [comment]
I understand it's the source file -- but my source field keeps coming up blank?

![image](https://user-images.githubusercontent.com/446329/51662653-a74a4380-2008-11e9-9304-27900272e37f.png)

**Comment by nijel** [comment]
Ah, you're right, I've fixed this in 48abae2794.
